modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/Loader.scala

```diff
@@ -105,16 +105,44 @@ object Loader {
 
     def initRetry(f: F[Unit]) = Retry.retryingOnAllErrors(config.initRetries, initRetryLog[F], f)
 
+    def isPermissionError(t: Throwable): Boolean = {
+      val err = t.toString.toLowerCase
+      err.contains("insufficient privileges") || // for Snowflake
+      err.contains("invalid operation: permission denied") // for Redshift
```

Comment:
Is it possible to pattern match on, say, the SqlState or vendor code of the SqlException?  If it's not possible then I'm OK with how you've done it here.